### PR TITLE
Rename backends

### PR DESCRIPTION
**Pull request by jdahm**
## Description

- Remove `gtc:` prefix on backend names
- Merge `src/gt4py/backend/gtc_backend` into `src/gt4py/backend`

~Blocked by #698.~

Part of #566.

### CODE REVIEW

**Comment by jdahm** [comment]
bors try

**Comment by jdahm** [comment]
bors try-
bors try

File: docs/gt4py/quickstart.rst [hunk 1]
```diff
@@ -128,10 +128,10 @@ function with the ``stencil`` decorator provided by GT4Py.
...
 
-* ``"gtc:numpy"``: a vectorized python backend
-* ``"gtc:gt:cpu_kfirst"``: a backend based on GridTools code performance-optimized for x86 architecture
-* ``"gtc:gt:cpu_ifirst"``: a GridTools backend targeting many core architectures
-* ``"gtc:gt:gpu"``: a GridTools backend targeting GPUs
+* ``"numpy"``: a vectorized python backend
+* ``"gt:cpu_kfirst"``: a backend based on GridTools code performance-optimized for x86 architecture
+* ``"gt:cpu_ifirst"``: a GridTools backend targeting many core architectures
+* ``"gt:gpu"``: a GridTools backend targeting GPUs
 * ``"gtc:gpu"``: a backend targeting NVIDIA GPUs using CUDA and some GridTools
```
Comment (gronerl) [inline docs/gt4py/quickstart.rst:135]: This should have said `"gtc:cuda"` before, now `"cuda"`?

File: src/gt4py/backend/gtc_common.py [hunk 2]
```diff
@@ -261,14 +261,14 @@ def generate_computation(self) -> Dict[str, Union[str, Dict]]:
...
+        self, language_name: str, *, stencil_ir: gtir.Stencil = None
     ) -> Dict[str, Union[str, Dict]]:
-        if not ir:
-            ir = self.builder.gtir
+        if not stencil_ir:
+            stencil_ir = self.builder.gtir
         if language_name != "python":
             return super().generate_bindings(language_name)
         dir_name = f"{self.builder.options.name}_src"
-        src_files = self.make_extension_sources(stencil_ir=ir)
```
Comment (gronerl) [inline src/gt4py/backend/gtc_common.py:271]: Is this an unrelated change? (Not against it...)

File: src/gt4py/storage/utils.py [hunk 3]
```diff
@@ -48,7 +48,7 @@ def normalize_storage_spec(default_origin, shape, dtype, mask):
             - default_origin: tuple of ints with default origin values for the non-masked dimensions
             - shape: tuple of ints with shape values for the non-masked dimensions
             - dtype: scalar numpy.dtype (non-structured and without subarrays)
-            - backend: backend identifier string (gtc:numpy, gtc:gt:cpu_kfirst, gtc:gpu, ...)
+            - backend: backend identifier string (numpy, gt:cpu_kfirst, gtc:gpu, ...)
```
Comment (gronerl) [inline src/gt4py/storage/utils.py:51]: `"gtc:gpu"` is not a backend anymore, assuming my comment above makes sense

**Comment by gronerl** [comment]
Should we also remove the prefix in file names (`gtc_common.py` or classes (`GTCDace?PUBackend` etc)?

File: src/gt4py/backend/gtc_common.py (hunk 2 above)
Comment (jdahm) [reply src/gt4py/backend/gtc_common.py:271]: The previous PR was supposed to rename all the `ir` to `stencil_ir`, but I must have forgotten this one :).

File: docs/gt4py/quickstart.rst (hunk 1 above)
Comment (jdahm) [reply docs/gt4py/quickstart.rst:135]: Good catch! Changed.

**Comment by jdahm** [comment]
bors try

**Comment by gronerl** [comment]
bors try

**Review by gronerl — APPROVED** [review]
